### PR TITLE
Update Zowe CLI dependency to 7.27.0

### LINKED ISSUE #3003: Update Zowe SDKs to 7.27.0 for proxy support
**Problem**
In Zowe CLI 7.27.0, proxy support was added using the `http-proxy-agent` package. This package compiles to an ES2020 target and its source code contains optional chaining which is only supported in ES2020 and up. Since Webpack 4 which we still use for ZE V2 does not support ES2020 syntax, ZE builds fail after updating Zowe CLI to the latest version.

**Solution**
Investigate a way to fix the Webpack build so that Zowe CLI can be updated. We may need to configure a Babel loader that converts ES2020 code to be compatible with an older target like ES6.

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

Resolves #3003

## Release Notes

Milestone: 2.17.0

Changelog: Updated the `@zowe/cli` dependency to 7.27.0 to support proxy environment variables

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [ ] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Review by traeok — APPROVED** [review]
With the Babel changes, I was able to build ZE without errors and search data sets as expected.

Thanks @t1m0thyj!

File: packages/zowe-explorer-api/CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,8 @@ All notable changes to the "zowe-explorer-api" extension will be documented in t
 
 ### New features and enhancements
 
+- Updated the `@zowe/cli` dependency to 7.27.0 to support proxy environment variables [#3003](https://github.com/zowe/zowe-explorer-vscode/issues/3003)
```
Comment (anaxceron) [inline packages/zowe-explorer-api/CHANGELOG.md:9]: should this say **v** 7.27.0?

**Review by anaxceron — APPROVED** [review]
Minor question, but otherwise good to go.

File: packages/zowe-explorer-api/CHANGELOG.md (hunk 1 above)
Comment (t1m0thyj) [reply packages/zowe-explorer-api/CHANGELOG.md:9]: I modeled it after previous changelog entries but can change it if the "v" prefix is preferred 😋

**Review by JillieBeanSim — APPROVED** [review]
LGTM! thanks @t1m0thyj

**Pull request merged**